plugins/rocksdb/src/main/java/org/hyperledger/besu/plugin/services/storage/rocksdb/segmented/RocksDBColumnarKeyValueStorage.java

```diff
@@ -148,7 +148,7 @@ public RocksDBColumnarKeyValueStorage(
   }
 
   private BlockBasedTableConfig createBlockBasedTableConfig(final RocksDBConfiguration config) {
-    final LRUCache cache = new LRUCache(config.getCacheCapacity());
+    final LRUCache cache = new LRUCache(config.getCacheCapacity(), -1, true);
```

Comment:
Wouldn't we also need to handle the new read failures we could now get if the cache is full? We'd need to have something that provides back-pressure to limit the rate of reads we attempt until the DB has cache available again and can serve them. Otherwise we'll wind up at best spamming the logs with scary exceptions as reads fail and possibly wind up rejecting blocks we should accept and causing a chain fork.

Reply:
Good point, but my understanding is that when the cache is full and the insertion is rejected, it is handled internally and the KV entry is retrieved from SST files. I think the side effect here is more a performance degradation because of the cache misses and more IO to retrieve these missed entries from disk (SST files).

Reply:
Some errors happens and the nodes 2 of the four nodes crashed.
By enabling Jemalloc and setting block cache without forcing it resolved the native memory issue, I'm going then to close this one.